### ISSUE TITLE
Questions about the future of the project

### ISSUE THREAD

**Issue by patrickReiis** [BrianWill/japanese_vocab#4]
I did the pull request #3 with the purpose of making it easier for beginners with less experience in programming and also for people who do not want to type the full URL in the browser. However, you didn't give any feedback as to why you don't want to merge it. This made me think about a couple things.

- What is the end goal of this project?
- Do you pretend to grow this project to be used for a lot of people or is it just a hobby of yours?
- Do you pretend to make the project more coupled with the code aspect of it? For example you log messages in the console but if you intend to make this project more beginner friendly it would be good to display the errors on the website itself instead of only the standard output (I could code that for you).

I also want to say for the record that I think you’re perhaps one of the best programmers in the world and I want to be a programmer just like you. I’m still a beginner and I dedicate myself to code in my free time. So my last question is **on which specific part of the project you’d like help from other people**?

**Comment by BrianWill** [comment]
Hi Patrick, sorry I just haven't been paying attention to activity here. Merged #3 just now.

> I also want to say for the record that I think you’re perhaps one of the best programmers in the world

😆 Have you looked at the code? Thanks for the vote of confidence though.

The answer to most of your questions is "I'm being sloppy and lazy". Errors, for example, should in many cases be properly displayed to users, so yes that would be a welcome contribution. Another big issue is the lack of backend tests and database migration. Right now I'm just committing my own copy of the db, and so user data is getting clobbered anytime someone pulls. So separating out a test DB would be another good contribution.

In general though I'm still figuring out how to make the tool actually useful while keeping it very simple. I recently simplified how stories are tracked: users just group their stories into one of four categories. I also changed my mind about how to handle vocab tracking: users individually set a "countdown max" of 0, 2, or 5 (by hitting q, w, e), and then when the "reset" a story, the countdown of each word in the story is reset to its countdown max. Anyway, I should definitely update the README to make this all more clear, and there should probably be on screen hints about the controls.
